### PR TITLE
Add a new endpoint to update the isCompleted status of a todo item

### PR DESCRIPTION
**Pull request by yamin-faruque-sazim**
**DESCRIPTION OF CHANGES**

- Define a new patch endpoint in the controller to update the status 
- Update default sort to consider isCompleted status, so that completed items are sorted to the bottom

### CODE REVIEW

File: src/todos/todos.repository.ts [hunk 1]
```diff
@@ -15,13 +15,24 @@ export class TodoRepository extends EntityRepository<Todo> {
...
   }
+  // async findAllSorted(
+  //   sortBy: string = 'createdAt',
+  //   sortOrder: string = 'asc',
+  // ): Promise<Todo[]> {
+  //   const order = sortOrder === 'desc' ? 'DESC' : 'ASC';
+  //   return await this.findAll({
+  //     orderBy: { [sortBy]: order },
+  //   });
+  // }
```
Comment (ishmam-iqbal-sazim) [inline src/todos/todos.repository.ts]: Let's remove comments

File: src/todos/todos.service.ts [hunk 2]
```diff
@@ -37,4 +37,13 @@ export class TodosService {
     const todoItem = await this.getTodoById(id);
     await this.todosRepository.getEntityManager().removeAndFlush(todoItem);
   }
+
+  async updateTaskCompletion(id: string, isCompleted: boolean): Promise<Todo> {
+    const task = await this.todosRepository
+      .getEntityManager()
+      .findOneOrFail(Todo, { id });
```
Comment (ishmam-iqbal-sazim) [inline src/todos/todos.service.ts]:
```suggestion
    const task = await this.todosRepository.findOneOrFail({ id });
```
Can't we do this?

**Review by ishmam-iqbal-sazim — APPROVED** [review]
Approved pending comments

File: src/todos/todos.service.ts (hunk 2 above)
Comment (yamin-faruque-sazim) [reply src/todos/todos.service.ts]: Yes, addressed in latest commit

**Pull request merged**